src/server_manager/web_app/app.ts

```diff
@@ -289,319 +312,197 @@ export class App {
     });
 
-    appRoot.addEventListener('ShowServerRequested', (event: CustomEvent) => {
-      this.handleShowServerRequested(event.detail.displayServerId);
+    appRoot.addEventListener('ShowServerRequested', async (event: CustomEvent) => {
+      const server = this.getServerById(event.detail.displayServerId);
+      if (server) {
+        await this.showServer(server);
+      } else {
+        console.error(
+            `Could not find server for display server ID ${event.detail.displayServerId}`);
+      }
     });
 
     onUpdateDownloaded(this.displayAppUpdateNotification.bind(this));
   }
 
-  async start(): Promise<void> {
-    this.showIntro();
-    await this.syncDisplayServersToUi();
-
-    const manualServersPromise = this.manualServerRepository.listServers();
-
-    const accessToken = this.digitalOceanTokenManager.getStoredToken();
-    const managedServersPromise = !!accessToken ?
-        this.enterDigitalOceanMode(accessToken).catch(e => [] as server.ManagedServer[]) :
-        Promise.resolve([]);
-
-    const managedServers = await managedServersPromise;
-    const serverBeingCreated = managedServers.find(server => !server.isInstallCompleted());
-    if (!!serverBeingCreated) {
-      this.syncServerCreationToUi(serverBeingCreated);
-    }
-
-    const installedManagedServers = managedServers.filter(server => server.isInstallCompleted());
-    const manualServers = await manualServersPromise;
-    await this.syncServersToDisplay(manualServers.concat(installedManagedServers));
+  // Shows the intro screen with overview and options to sign in or sign up.
+  private showIntro() {
+    this.appRoot.showIntro();
+  }
 
-    await this.maybeShowLastDisplayedServer();
+  private displayAppUpdateNotification() {
+    this.appRoot.showNotification(this.appRoot.localize('notification-app-update'), 60000);
   }
 
-  private async syncServersToDisplay(servers: server.Server[]) {
-    for (const server of servers) {
-      await this.syncServerToDisplay(server);
-    }
+  async start(): Promise<void> {
+    this.showIntro();
 
-    // Remove any unsynced servers from display and alert the user.
-    const displayServers = await this.displayServerRepository.listServers();
-    const unsyncedServers = displayServers.filter(s => !s.isSynced);
-    if (unsyncedServers.length > 0) {
-      for (const displayServer of unsyncedServers) {
-        this.displayServerRepository.removeServer(displayServer);
-      }
-      const unsyncedServerNames = unsyncedServers.map(s => s.name).join(', ');
-      let messageKey = 'error-server-removed';
-      let placeholder = 'serverName';
-      if (unsyncedServers.length > 1) {
-        // Pluralize localized message.
-        messageKey = 'error-servers-removed';
-        placeholder = 'serverNames';
+    // Load server list. Fetch manual and managed servers in parallel.
+    await Promise.all([
+      this.loadDigitalOceanServers(this.digitalOceanTokenManager?.getStoredToken()),
```

Comment:
Same: optional chaining likely not needed.

Reply:
Keeping as I explained before. It makes it clearer we don't need to have the token manager.